contributed_definitions/nyaml/NXdata_photoemission.yaml

```diff
@@ -0,0 +1,1143 @@
+        Do not use the ``axes`` attribute with the ``axis`` attribute.
+    \@long_name:
+      doc: |
+        data label
+  errors(NX_NUMBER):
+    deprecated: Use ``DATA_errors`` instead (NIAC2018)
+    doc: |
+      Standard deviations of data values -
+      the data array is identified by the group attribute ``signal``.
+      The ``errors`` array must have the same dimensions as ``DATA``.
+      Client is responsible for defining the dimensions of the data.
+    dimensions:
+      rank: dataRank
+      doc: |
+        The ``errors`` must have
+        the same rank (``dataRank``)
+        as the ``data``.
+        At least one ``dim`` must have length "n".
+      dim: []
+  scaling_factor(NX_FLOAT):
+    doc: |
+      The elements in data are usually float values really. For
+      efficiency reasons these are usually stored as integers
+      after scaling with a scale factor. This value is the scale
+      factor. It is required to get the actual physical value,
+      when necessary.
+  offset(NX_FLOAT):
+    doc: |
+      An optional offset to apply to the values in data.
+  title:
+    doc: |
+      Title for the plot.
+  x(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      This is an array holding the values to use for the x-axis of
+      data.  The units must be appropriate for the measurement.
+    dimensions:
+      rank: 1
+      dim: [[1, nx]]
+  y(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      This is an array holding the values to use for the y-axis of
+      data.  The units must be appropriate for the measurement.
+    dimensions:
+      rank: 1
+      dim: [[1, ny]]
+  z(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      This is an array holding the values to use for the z-axis of
+      data.  The units must be appropriate for the measurement.
+    dimensions:
+      rank: 1
+      dim: [[1, nz]]
+  energy(NX_NUMBER):
+    unit: NX_ENERGY
+    doc: |
+      Calibrated energy axis.
+      
+      In an application definition, this could be a link to either
+      /entry/process/energy_calibration/calibrated_axis or
+      /entry/process/energy_correction/calibrated_axis.
+    \@type:
+      type: NX_CHAR
+      doc: |
+        The energy can be either stored as kinetic or as binding energy.
+      enumeration:
+        kinetic:
+          doc:
+          - |
+            Calibrated kinetic energy axis.
+          - |
+            xref:
+              spec: ISO 18115-1:2023
+              term: 3.35
+              url: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:3.35
+        binding:
+          doc:
+          - |
+            Calibrated binding energy axis.
+          - |
+            xref:
+              spec: ISO 18115-1:2023
+              term: 12.16
+              url: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.16
+  \@energy_indices:
+  \@energy_depends:
+  momentum(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Calibrated momentum axis.
+      Units are kg*m/s.
```

Comment:
These should also have inverse real space units, i.e. NX_WAVENUMBER, no?

Reply:
Right.

One question as I am not doing ARPES: is there any difference between momentum and the individual components in k-space? Maybe they could be stored together?

Reply:
Well, I don't actually know where this is one here is coming from. One can define a general k_parallel, that comprises k_x and k_y. k_z is typically taken as perpendicular component. Also one can consider internal and external momenta, where only the parallel ones are conserved

Reply:
It came from the fact that we had a `momentum_calibration` in `NXmpes`. To me, it is fine to remove `NXprocess_photoemission/momentum_calibration` `NXdata_photoemission/momentum` again and just keep the k-axes?

We can add your description of the parallel/perpendicular components to the different k-axes.

Reply:
How about the following then?

**NXdata_photoemission:** (similarly in NXdata_photoemission_detector)
```yaml
kN(NX_NUMBER):
  unit: NX_WAVENUMBER
  doc: |
    One calibrated k-space coordinate. It is envisioned that the axes are named kx, ky, and kz,
    in accordance with the calibrations defined in NXprocess_photoemission.

    Typically, the vectors in momentum space are defined such that kx and ky comprise the parallel component,
    while kz is taken as the perpendicular component.

    It is also possible to define k_parallel and k_perp for the parallel and perpendicular momenta, respectively.
    Units are 1/Angström.
```

**NXprocess_photoemission:**
```yaml
kN_calibration(NXcalibration):
  doc: |
    Calibration event on a k-space axis.
    
    It is envisioned that the individual calibrations for each coordinate are
    stored as kx_calibration, ky_calibration, kz_calibration, in accordance
    with the axis name definitions in NXdata_photoemission.

    It is also possible to have k_parallel_calibration and k_perp_calibration if
    the momentum axes are defined as k_parallel and k_perp for the parallel and
    perpendicular momenta, respectively.
  physical_quantity:
    enumeration: [momentum]
  calibrated_axis(NX_FLOAT):
    unit: NX_WAVENUMBER
    doc: |
      This is the calibrated k-space axis to be used for data plotting.
```